nixos/modules/services/networking/nncp.nix

```diff
@@ -0,0 +1,151 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  statefulCfgFile = "/etc/nncp.secrets";
```

Comment:
Whenever I see `stateful` and `/etc` I wonder if there is a different way to do things - a way that better represents nix philosophies.

Maybe we should create a `secretsFile` option with `type = with types; nullOr path;` which `default`s to `null`. If the sysadmin provides a file then we use it, otherwise we generate our own and store it in a stateful directory, like `/var/lib/nncp/config`. If we pair this with [LoadCredential](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#LoadCredential=ID:PATH) I think we have a reasonable chance at running this service with `DynamicUser = true;`.

Reply:
This is complicated because `/etc/nncp.hjson` must contain private keys and configuration, but nodes can only connnect to each other when the public key is known. It may be better to use a [configuration directory](http://www.nncpgo.org/Configuration-directory.html).